Aplib.Tests/Desire/GoalTests.cs

```diff
@@ -0,0 +1,84 @@
+using Aplib.Core.Desire;
+using Aplib.Tests.Stubs.Desire;
+using FluentAssertions;
+
+namespace Aplib.Tests.Desire;
+
+public class GoalTests
+{
+    [Fact]
+    public void Goal_WhenConstructed_ContainsCorrectMetaData()
```

Comment:
A comment for every tests would be nice, describing the test in the style of Given, When, Then.

Example:
```suggestion
    /// <summary>
    /// Given valid parameters and metadata,
    /// When the goal is constructed,
    /// Then the goal should correctly store the metadata.
    /// </summary>
    [Fact]
    public void Goal_WhenConstructed_ContainsCorrectMetaData()
```